### PR TITLE
Add support for tracing via OpenTelemetry

### PR DESCRIPTION
**Pull request by josephwoodward**
As OpenTracing (the predecessor to OpenTelemetry) has been deprecated, this change adds support for OpenTelemetry via the `WithOpenTelemetry` functional option.

A lot of the API and options align with the [`otelhttp`](https://github.com/open-telemetry/opentelemetry-go-contrib/blob/main/instrumentation/net/http/otelhttp/transport.go) transport that can be used on the net/http client 

Addresses issue https://github.com/go-openapi/runtime/issues/249

### CODE REVIEW

File: client/opentelemetry.go [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+		op.Context = ctx
+
+		span.SetAttributes(
+			attribute.String("net.peer.name", host),
+			// attribute.String("net.peer.port", ""),
+			attribute.String(string(semconv.HTTPRouteKey), op.PathPattern),
+			attribute.String(string(semconv.HTTPMethodKey), op.Method),
+			attribute.String("span.kind", trace.SpanKindClient.String()),
+			//TODO: Figure out the best way to get the scheme?
+			attribute.String("http.scheme", op.Schemes[0]),
```
Comment (josephwoodward) [inline client/opentelemetry.go:86]: Hi @casualjim, can you offer any pointers on the best way to get the scheme? I don't appear to have access to the underlying `http.Request` object to check the existence of the `TLS` field (which is one common way to work out the scheme)

**Comment by casualjim** [comment]
Thanks for all your work on this.

**Review by casualjim — APPROVED** [review]
This looks good.

I'll see if I can figure out how to get the actual scheme that's being used for the request into here too.

**Comment by casualjim** [comment]
can you bump the go version in the .github/ folder to be at least 1.18 or higher? Otel seems to require it

**Comment by josephwoodward** [comment]
@casualjim That's been bumped, do you want me to bump the version in `go.mod` too?

**Comment by casualjim** [comment]
yeah that would be good

**Comment by yvvq** [comment]
@casualjim  Could you please merge this pr if It’s ok with you.

**Pull request merged**